### PR TITLE
Add nightlies for Nomad

### PR DESCRIPTION
**Pull request by claire-labry**
This is a draft PR for the nightly releases for the OSS branch. @sarahethompson  and I have partially tested this workflow, however we would like to double check and see if it’s ok to test a full nightly release/tag? @lgfa29 

Testing has been completed! Here's the successful workflow run: https://github.com/hashicorp/nomad/actions/runs/2196537794

The successful nightly release is here: https://github.com/hashicorp/nomad/releases [Note: the test nightly release will be cleaned up and removed once all looks good]

A `SLACK_WEBHOOK_URL` has been added and it will only trigger a slack message on failure. The assumption is that if the nightly release is successful, the `slack-notify` step will be skipped. The Nomad team can configure this preference and change it to `always` or whatever suits your team the best! The messages are sent to the `#feed-nomad-releases` channel. 

Please let us know if you have any questions!

### CODE REVIEW

**Comment by lgfa29** [comment]
Yup, feel free to create them for testing.

File: .github/workflows/nightly-release.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  schedule:
+    # Runs against the default branch every day at midnight
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+
+jobs:
+  # Build a fresh set of artifacts
+  build-artifacts:
+    # TODO: change branch from add-nightlies to main before merging this PR
+    uses: hashicorp/nomad/.github/workflows/build.yml@add-nightlies
```
Comment (sarahethompson) [inline .github/workflows/nightly-release.yml:20]: Ah we'll need to switch this back to main!

**Review by sarahethompson — APPROVED** [review]

File: .github/workflows/nightly-release.yml (hunk 1 above)
Comment (claire-labry) [reply .github/workflows/nightly-release.yml]: Great eye! Thank you @sarahethompson ❤️

File: .github/workflows/nightly-release.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  # TODO: remove the push event before merging this PR; this is for testing purposes
+  schedule:
+    # Runs against the default branch every day at midnight
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+
+jobs:
+  # Build a fresh set of artifacts
+  build-artifacts:
+    # TODO: change branch from add-nightlies to main before merging this PR
```
Comment (lgfa29) [inline .github/workflows/nightly-release.yml:19]:
```suggestion

```

Also done ✅

File: .github/workflows/nightly-release.yml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+          allowUpdates: true
+          removeArtifacts: true
+          draft: false
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Publish nightly GitHub prerelease
+        uses: eregon/publish-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          release_id: ${{ steps.create_prerelease.outputs.id }}
```
Comment (lgfa29) [inline .github/workflows/nightly-release.yml:78]: Out of curiosity, what does this action do? I read their docs but still could quite get what "publish the release" means in this case.

**Review by lgfa29 — APPROVED** [review]
This is great, thanks for putting it together!

File: .github/workflows/nightly-release.yml [hunk 4]
```diff
@@ -0,0 +1,111 @@
+# This GitHub action triggers a fresh set of Nomad builds
+# and publishes them to GitHub Releases under the `nightly` tag.
+# Note that artifacts available via GitHub Releases are not codesigned or notarized.
+# Failures are reported to slack.
+#
+
+name: Nightly Release
+
+on:
+  # TODO: remove the push event before merging this PR; this is for testing purposes
```
[the remark this answers is not in the document]
Comment (claire-labry) [reply .github/workflows/nightly-release.yml:10]: facepalm (and the other TODO comment) thank you so much for catching these! 😅

File: .github/workflows/nightly-release.yml [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  schedule:
+    # Runs against the default branch every day at midnight
+    - cron: "0 0 * * *"
```
[the remark this answers is not in the document]
Comment (claire-labry) [reply .github/workflows/nightly-release.yml:13]: That makes total sense. ❤️

File: .github/workflows/nightly-release.yml (hunk 3 above)
Comment (claire-labry) [reply .github/workflows/nightly-release.yml:76]: I believe this action takes the `nightly` Github prerelease (previous action) and communicates with Github and automates the process of pushing a nightly release + tag for users to utilize. cc: @sarahethompson to correct/clarify me if I'm wrong!

**Comment by mdeggies** [comment]
Thoughts on closing this? While Packer and a couple of other teams have adopted this workflow, these nightlies don't make it through the entire CRT pipeline, meaning artifacts aren't scanned, codesigned or notarized. :( We've seen little value-add to publishing nightly github releases with the raw artifacts from the build workflow that haven't been properly processed.  We do want to support this long term, and are scoping this out over in https://hashicorp.atlassian.net/browse/HREL-28

**Comment by tgross** [comment]
Hi @mdeggies!

> Thoughts on closing this? While Packer and a couple of other teams have adopted this workflow, these nightlies don't make it through the entire CRT pipeline, meaning artifacts aren't scanned, codesigned or notarized. 

That's unfortunate... artifacts that were codesigned would be hugely valuable for our OSS community I think. We currently build nightlies as part of E2E but they're all just "built locally" as part of the E2E testing workflow, so we can't ship those to the community. I'm :+1: for closing this if we've got another effort underway though.

**Comment by claire-labry** [comment]
@mdeggies I'll close this today in favor of https://hashicorp.atlassian.net/browse/HREL-28  👍🏻

**Pull request closed without merging**